html/browsers/origin/api/origin-from-worker.window.js

```diff
@@ -0,0 +1,31 @@
+// META: title=`Origin.from(Worker)`
+
+const src = `
+const originA = Origin.from(globalThis);
+const originB = Origin.from(globalThis);
+
+self.postMessage({
+  "isOpaque": originA.opaque,
+  "sameOrigin": originA.isSameOrigin(originB),
+});
+`;
+
+test(t => {
+  const dataURL = `data:text/html;base64,${btoa(src)}`;
+  const worker = new Worker(dataURL);
+  worker.onmessage = e => {
+    assert_true(e.data.isOpaque, "Origin created from data URL Worker should be an opaque origin.");
+    assert_false(e.data.sameOrigin, "Two data URL opaque origins should not be same-origin with one another.");
```

Comment:
My original interpretation of https://html.spec.whatwg.org/multipage/workers.html#set-up-a-worker-environment-settings-object of returning a new opaque origin every time doesn't seem right, based on test results maybe I should make a MR to the HTML spec to change it to be more like the [Worklet](https://html.spec.whatwg.org/multipage/worklets.html#set-up-a-worklet-environment-settings-object) case which stores origin as a variable instead of creating a new one each time.